### ISSUE TITLE
There are 403 logs after deployment

### ISSUE THREAD

**Issue by guleng** [DirectXMan12/k8s-prometheus-adapter#274]
```
I0302 07:57:40.557744       1 round_trippers.go:438] GET https://10.68.0.1:443/api/v1/pods?resourceVersion=1483856&timeout=8m9s&timeoutSeconds=489&watch=true 403 Forbidden in 2 milliseconds
I0302 07:57:40.557792       1 round_trippers.go:444] Response Headers:
I0302 07:57:40.557801       1 round_trippers.go:447]     Content-Type: application/json
I0302 07:57:40.557806       1 round_trippers.go:447]     X-Content-Type-Options: nosniff
I0302 07:57:40.557812       1 round_trippers.go:447]     Content-Length: 299
I0302 07:57:40.557817       1 round_trippers.go:447]     Date: Mon, 02 Mar 2020 07:57:40 GMT
I0302 07:57:40.557970       1 request.go:947] Response Body: {"kind":"Status","apiVersion":"v1","metadata":{},"status":"Failure","message":"pods is forbidden: User \"system:serviceaccount:custom-metrics:custom-metrics-apiserver\" cannot watch resource \"pods\" in API group \"\" at the cluster scope","reason":"Forbidden","details":{"kind":"pods"},"code":403}
E0302 07:57:40.558086       1 reflector.go:283] k8s.io/client-go/informers/factory.go:133: Failed to watch *v1.Pod: unknown (get pods)
I0302 07:57:40.630759       1 reflector.go:160] Listing and watching *v1.Node from k8s.io/client-go/informers/factory.go:133
I0302 07:57:40.631215       1 round_trippers.go:419] curl -k -v -XGET  -H "Accept: application/json, */*" -H "User-Agent: adapter/v0.0.0 (linux/amd64) kubernetes/$Format" -H "Authorization: Bearer eyJhbGciOiJSUzI1NiIsImtpZCI6IiJ9.eyJpc3MiOiJrdWJlcm5ldGVzL3NlcnZpY2VhY2NvdW50Iiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9uYW1lc3BhY2UiOiJjdXN0b20tbWV0cmljcyIsImt1YmVybmV0ZXMuaW8vc2VydmljZWFjY291bnQvc2VjcmV0Lm5hbWUiOiJjdXN0b20tbWV0cmljcy1hcGlzZXJ2ZXItdG9rZW4tZHNkcnMiLCJrdWJlcm5ldGVzLmlvL3NlcnZpY2VhY2NvdW50L3NlcnZpY2UtYWNjb3VudC5uYW1lIjoiY3VzdG9tLW1ldHJpY3MtYXBpc2VydmVyIiwia3ViZXJuZXRlcy5pby9zZXJ2aWNlYWNjb3VudC9zZXJ2aWNlLWFjY291bnQudWlkIjoiZDVjY2VlM2QtNWM1My0xMWVhLWJhMWItZmExNjNlZWMwYzg4Iiwic3ViIjoic3lzdGVtOnNlcnZpY2VhY2NvdW50OmN1c3RvbS1tZXRyaWNzOmN1c3RvbS1tZXRyaWNzLWFwaXNlcnZlciJ9.B8TXbIa3NGvaCxW93anFsziH0FZPwGa64MYF1BkH-Z78Yzwt574jinf5Ai17UT0lpxP96xcuWOJ6vrGlfzHOZ298QcxMr8XIjCRH7xqYN1c6LfiAa18p0v4a4StO28vlHUVtykQ6LLIwEn5GLGUN4REBZ8iXS2KNoVt0dV_0JKpKdF1WYtF5vu2kEKDliNRN4G94ddcdjk8B4AH9ks5xS4rS5-kzzkP7Re4C5sW9RgbwRigndnMjbsOollkdDGg0GK1oAxsy7cY-GIwSr8Ey2aajfgVJ3QoibNY0rh4I0wukw-Gk3AnP2HReqhE2qnQNTMe_3i5OoB5E16nFSatepA" 'https://10.68.0.1:443/api/v1/nodes?limit=500&resourceVersion=0'

```
I don't know how to deal with these 403. There are a lot of them. Are they certificate issues or RBAC permissions

**Comment by GMartinez-Sisti** [comment]
The error message is right there:

`User \"system:serviceaccount:custom-metrics:custom-metrics-apiserver\" cannot watch resource \"pods\" in API group \"\" at the cluster scope","reason":"Forbidden"`

I would double check if the role bindings, etc, are correctly applied. Check here for the yaml files-> https://github.com/DirectXMan12/k8s-prometheus-adapter/tree/master/deploy/manifests

**Comment by nirrozenbaum** [comment]
hi I'm seeing the exact same error both for pods and for nodes.
I've checked the yaml files and all of them match exactly to the ones that are located in the repo.
any idea what could be wrong?

**Comment by GMartinez-Sisti** [comment]
@nirrozenbaum Where is your kubernetes cluster hosted?

I'm not sure if a PAAS solution such as (EKS, GKE or AKS) allows you to get that type of access.

Also confirm if the deployed resources are correct with `kubectl get <type> <name> -o yaml`.

**Comment by nirrozenbaum** [comment]
I found the issue, I fixed the ``` cannot watch resource``` problem by adding the verb watch in the cluster role yaml file.

**Comment by GMartinez-Sisti** [comment]
I see that the examples in this repository don't contain the verb `watch` and even the resource `nodes` ([here](https://github.com/DirectXMan12/k8s-prometheus-adapter/blob/master/deploy/manifests/custom-metrics-resource-reader-cluster-role.yaml)).

But the helm repository does ([here](https://github.com/helm/charts/blob/master/stable/prometheus-adapter/templates/resource-metrics-cluster-role.yaml)) which means they have to be updated.

@nirrozenbaum are you willing to send a PR with this change?

**Comment by nirrozenbaum** [comment]
@GMartinez-Sisti sure, I'll create a PR with the updated role.

**Comment by nirrozenbaum** [comment]
btw, I think also [this file](https://github.com/helm/charts/blob/master/stable/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml) has to be updated as well, isn't it?

**Comment by GMartinez-Sisti** [comment]
Looks like it. Nice catch!

**Comment by yashbhutwala** [comment]
> btw, I think also [this file](https://github.com/helm/charts/blob/master/stable/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml) [](/helm/charts/blob/HEAD@{2020-03-23T07:22:05Z}/stable/prometheus-adapter/templates/custom-metrics-resource-reader-cluster-role.yaml) has to be updated as well, isn't it?

fixed this in the helm chart 😄

**Comment by nirrozenbaum** [comment]
created pull request to fix the issue.

**Comment by s-urbaniak** [comment]
fixed in #274
